### ISSUE TITLE
should not create correlation buffer if the feature is not supported

### ISSUE THREAD

**Issue by XinfengZhang** [intel/libva-utils#104]
test CreateBuffer/VAAPICreateBuffer.CreateBufferWithOutData/59
it will create VAMacroblockParameterBufferType buffer, but if driver dose not  support macroblock parameter, it should return failed instead of VA_STATUS_SUCCESS

**Comment by XinfengZhang** [comment]
also VAQMatrixBufferType in "CreateBuffer/VAAPICreateBuffer.CreateBufferWithOutData/61 - 63"

**Comment by uartie** [comment]
@XinfengZhang how do clients/apps detect if driver supports these buffer types or not?  Are there specific VA APIs that relay this information?

**Comment by XinfengZhang** [comment]
no vaapi to export the information, but we should not expect all buffer type can be created. so return unsupport error should also be acceptable.

**Comment by uartie** [comment]
Is there a set of buffer types that we know are "required" to be supported by all drivers if codec is supported?  I don't know if it's a good idea to just accept unsupported as a valid result for just any buffer type.  @xhaihao any comment?

**Comment by XinfengZhang** [comment]
@xhaihao any comments?

**Comment by xhaihao** [comment]
I agree the driver may return VA_STATUS_ERROR_UNSUPPORTED_BUFFERTYPE if the type is not supported for a given profile/entrypoint pair. In addition the doc should have a clear requirement on the required buffer type for each profile/entrypoint pair. We may update the doc in the future release.
